codes/codes.go

```diff
@@ -0,0 +1,229 @@
+/*
+ * This file has been modified for use in flux, but the original
+ * is part of the grpc-go project.
+ *
+ * The original can be located here: https://github.com/grpc/grpc-go/blob/master/codes/codes.go
+ *
+ * Copyright 2014 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package codes defines the error codes used by flux.
+package codes
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type Code uint32
+
+const (
+	// OK is returned on success.
+	OK Code = 0
+
+	// Canceled indicates the operation was canceled (typically by the caller).
+	Canceled Code = 1
+
+	// Unknown error. An example of where this error may be returned is
+	// if a Status value received from another address space belongs to
+	// an error-space that is not known in this address space. Also
+	// errors raised by APIs that do not return enough error information
+	// may be converted to this error.
+	Unknown Code = 2
+
+	// InvalidArgument indicates client specified an invalid argument.
```

Comment:
This name has made no sense while I have been playing around with reporting errors. I'm renaming it to `Invalid` to indicate that the client gave us something that was not valid rather than the specific "invalid argument" which doesn't make sense in contexts where there is no argument or the argument is missing.